keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1007 @@
+
+* Extended resource support is not added just for easing the transition to DRA
+  for the short term. Its ease of use is one big advantage to keep it remaining
+  useful for the long term.
+
+* Device plugin API must not change. The existing device plugin drivers must
+  continue working without change.
+
+* DRA driver API must not change. Core Kubernetes (kube-scheduler, kubelet) is
+  preferred over DRA driver for any change needed to support the feature.
+
+* Keep advertising only extended resources backed by device plugin in `node.status.Capacity`
+  for Alpha. It will be revisited for Beta, based on Alpha feedback.
+
+### Non-Goals
+
+* Minimize kubelet or kube-scheduler changes. The feature requires necessary
+  changes in both scheduling and actuation.
+
+* One node has both extended resource backed by DRA, and the same named
+  extended resource backed by device plugin at the same time.
+
+## Proposal
+
+The basic idea is the following:
+
+1. Introduce an `extended resource backed by DRA` concept. It is like the current extended
+   resource backed by device plugin, in that, it has a string name, and a
+   discrete countable quantity. Its capacity can be derived from DRA
+   `ResourceSlice`, its consumption is specified through pod's extended
+   resource request.
+1. Introduce a field `ExtendedResourceName` to `DeviceClass` to allow cluster
+   administrators to treat certain class of devices as an extended resource.
+1. Introduce a special `ResourceClaim` object to keep track of device allocations. It is special only in the sense that it is created by the scheduler. No semantic changes are needed in the ResourceClaim API for it.
+   for all extended resource requests backed by DRA for a pod. kube-scheduler
+   uses DRA scheduling algorithm to fit pod's extended resource request to a
+   node that advertises the extended resource in DRA `ResorceSlice` or extended
+   resources backed by device plugin. When using DRA devices, it creates a
+   special `ResourceClaim` for the pod with the allocation result recording
+   which devices were picked. More details on this special `ResourceClaim`
+   follow below.  When using extended resources advertised for a node by device
+   plugin, the existing resource tracking reserves them.
+1. Introduce a field `ExtendedResourceClaimStatus` to pod's `Status`, such that:
+    - the kubelet can find the special `ResourceClaim` while looking for claims to prepare
+    - the kubelet can pass the devices to containers in the pod with the extended
+   resource requests, based on the container/extended resource to device
+   request mapping in the `ExtendedResourceClaimStatus`.
+
+Some quick clarifications around the basic concepts: extended resource backed by
+device plugin, extended resource backed by DRA, and dynamic resource.
+
+* extended resource backed by device plugin uses pod's
+  spec.containers[].resources.requests to request for resources, it consumes the capacity
```

Comment:
@tallclair, does IPPR allow mutation of extended resources in `spec.containers[].resources.requests`?

Since ResourceClaims are immutable, it doesn't seem like there's a reasonable way to honor resize requests on DRA-backed extended resources.

What would happen with those resize requests on the kubelet with device-plugin backed things today?

Can we make sure the DRA-backed ones respond in a way that is not *worse* than what device-plugin backed things do?

Reply:
Nevermind, I just swept the IPPR validation and it limits resize to CPU/memory 👍 

@tallclair, can you make an explicit note in the IPPR design that we should consider this interaction if there was ever a desire to let extended resources be mutable.